### PR TITLE
Issue #4920 Restore ability to invalidate sessions on shutdown

### LINKED ISSUE #4920: Restore ability to delete sessions on stop
Pre jetty 9.4 there was the ability to have all sessions invalidated when the server stopped, we should restore this behaviour.

Also evaluate the consistency of treatment of the session data store by the session cache wrt expectation that the session data store can/cannot be null.

### PR DESCRIPTION
**Pull request by janbartel**
Closes #4920 

Pre jetty-9.4 HashSessionManager had the ability to invalidate all sessions when the server is shutdown, this PR adds this ability back for the DefaultSessionCache. Note it is meaningless for the NullSessionCache, because it never caches any sessions, so can never do any invalidations on shutdown.

### CODE REVIEW

**Comment by joakime** [comment]
How will this behave in a cloud session scenario?
Will Jetty now attempt to invalidate all sessions on the cloud too?

**Comment by janbartel** [comment]
Each jetty instance will invalidate those sessions that are in it's cache: if you have configured a persistent backing session store eg files, jdbc, mongo, etc etc, then the invalidated sessions will also be removed from that store.

File: jetty-server/src/main/config/modules/session-cache-null.mod [hunk 1]
```diff
@@ -18,4 +18,4 @@ etc/sessions/session-cache-null.xml
 [ini-template]
 #jetty.session.saveOnCreate=false
 #jetty.session.removeUnloadableSessions=false
-#jetty.session.flushOnResponseCommit=false
+#jetty.session.flushOnResponseCommit=false
```
Comment (gregw) [inline jetty-server/src/main/config/modules/session-cache-null.mod:21]: missing EOL here

**Review by gregw — APPROVED** [review]
other than a tiny niggle LGTM

**Pull request merged**